plugins/rbac/src/components/ConditionalAccess/ConditionsFormRow.tsx

```diff
@@ -24,108 +43,540 @@ const useStyles = makeStyles(theme => ({
-};
-
 export const ConditionsFormRow = ({
   conditionRulesData,
   conditionRow,
   selPluginResourceType,
   criteria,
   onRuleChange,
   setCriteria,
-  setErrors,
+  handleSetErrors,
   setRemoveAllClicked,
 }: ConditionFormRowProps) => {
   const classes = useStyles();
   const theme = useTheme();
+  const [nestedConditionRow, setNestedConditionRow] = React.useState<
+    Condition[]
+  >([]);
+  const [notConditionType, setNotConditionType] =
+    React.useState<NotConditionType>('simple-condition');
 
-  const handleCriteriaChange = (val: string) => {
-    setCriteria(val);
-    switch (val) {
-      case criterias.condition: {
-        onRuleChange({
-          condition: {
-            rule: '',
-            resourceType: selPluginResourceType,
-            params: {},
-          },
-        });
+  React.useEffect(() => {
+    const nestedConditions: Condition[] = [];
+    const criteriaTypes = [criterias.allOf, criterias.anyOf, criterias.not];
+    switch (criteria) {
+      case criterias.allOf:
+        extractNestedConditions(
+          (conditionRow.allOf as Condition[]) || [],
+          criteriaTypes,
+          nestedConditions,
+        );
         break;
-      }
-      case criterias.allOf: {
-        onRuleChange({
-          allOf: [
-            {
-              rule: '',
-              resourceType: selPluginResourceType,
-              params: {},
-            },
-          ],
-        });
+      case criterias.anyOf:
+        extractNestedConditions(
+          (conditionRow.anyOf as Condition[]) || [],
+          criteriaTypes,
+          nestedConditions,
+        );
         break;
-      }
-      case criterias.anyOf: {
-        onRuleChange({
-          anyOf: [
-            {
-              rule: '',
-              resourceType: selPluginResourceType,
-              params: {},
-            },
+      case criterias.not:
+        if (
+          criteriaTypes.includes(Object.keys(conditionRow.not as Condition)[0])
+        ) {
+          nestedConditions.push(conditionRow.not as Condition);
+          setNotConditionType('nested-condition');
+        }
+        break;
+      default:
+        break;
+    }
+
+    setNestedConditionRow(nestedConditions);
+  }, [conditionRow, criteria]);
+
+  const handleCriteriaChange = (val: string) => {
+    handleSetErrors([], criteria, undefined, undefined, undefined, true);
+    setCriteria(val);
+
+    const defaultRule = getDefaultRule(selPluginResourceType);
+
+    const ruleMap = {
+      [criterias.condition]: { condition: defaultRule },
+      [criterias.allOf]: { allOf: [defaultRule] },
+      [criterias.anyOf]: { anyOf: [defaultRule] },
+      [criterias.not]: { not: defaultRule },
+    };
+
+    if (val === criterias.not) {
+      setNotConditionType('simple-condition');
+    }
+
+    const ruleChange = ruleMap[val];
+    if (ruleChange) {
+      onRuleChange(ruleChange);
+    }
+  };
+
+  const updateRules = (updatedNestedConditionRow: Condition[] | Condition) => {
```

Comment:
This can be moved out

Reply:
@divyanshiGupta This function uses `criteria`, `conditionRow` and updates form using `onRuleChange` all from `ConditionsFormRow`. It makes more sense to leave it in its context to me. How do you think?